components/Context/DdfInfo.js

```diff
@@ -24,9 +28,158 @@ export default class DdfInfo extends React.Component {
+    let tmpIssues = Object.assign({}, this.state.issues);
+    tmpIssues[this.state.selectedCategory].filters[name] = checked;
+    this.setState({issues: tmpIssues});
+  }
+
+  setCategory(selectedCategory){
+    if(this.state.selectedCategory !== selectedCategory){
+      this.setState({selectedCategory})
+    }
+  }
+
+ /*
+ * this method splits issues by categories and also sets it up so that
+ + its easy to handle the state of each filter
+ */
+ organizeIssues(issues){
+    let processedIssues = {};
+    let categories = [];
+
+    //isues that don't have a bounty
+    issues.filter(issue => issue.contractAddress === -1);
+
+    issues.forEach(issue => {
+      const category = issue.category;
+      if(!processedIssues[category]){
+        processedIssues[category] = {issues: [issue], filters:{}};
+        categories.push(category);
+      }
+      else{
+        processedIssues[category].issues.push(issue);
+      }
+      issue.labels.forEach(label => {
+        //if the user has toggled the filter off before then let it be
+        if(this.state.issues && this.state.issues[category] && this.state.isssues[category].filters[label] === false){
+          processedIssues[category].filters[label] = false;
+        }
+        else if(processedIssues[category].filters[label] === undefined){
+          //filters always start off as checked
+          processedIssues[category].filters[label] = true;
+        }
+      })
+    });
+
+    this.setState({issues: processedIssues, categories})
+ }
+
+ getContractAddress(issue, comments){
+  //TODO Change from owner to status' username once jose implements the end point
+  comments.filter(comment => comment.author_association === "OWNER");
+  let contractAddress = -1;
+  let match = comments[0].body.match(ethereumRegex());
+  if(match && match.length > 0){
+    contractAddress = match[0];
+  }
+  return {
+    ...issue,
+    contractAddress,
+    value: "$100"
+  };
+ }
+
+async getCommentsOfIssue(repoName, number){
+  return new Promise(async (resolve, reject) => {
+    axios.get(`https://api.github.com/repos/csmartinsfct/${repoName}/issues/${number}/comments?access_token=573dbede2e9766fd4d41610d15a8b82371fd522a`)
+      .then(function (response) {
+        // handle success
+       resolve(response.data);
+      })
+      .catch(function (error) {
+        // handle error
+        console.log(error);
+      })
+    })
+ }
+
+ getCategory(repoName){
+  if(repoName.indexOf("tech") !== -1 || repoName.indexOf("website") !== -1)
+    return "Development";
+  else if(repoName.indexOf("design") !== -1){
+    return "Design";
+  }
+  else if(repoName.indexOf("marketing") !== -1)
+    return "Marketing";
+  //TODO Remove after implementing jose's end point - just because my repos don't have these new names yet :x
+  else if(repoName === "MyBit-Go")
+    return "Development";
+  else
+    return "Design";
+}
+
+ async getIssues(){
+  try{
+    let issuesTmp = await Promise.all(Repos.map(async (repoName) =>
```

Comment:
No need to use `let`, your values aren't changing. Even if they were, we frown upon `let` because ideally our data never changes, we just create, modify and expand our data.

Reply:
I don't follow. I'm using let to initialize a variable to store the responses from the requests, which is then processed.

Reply:
I want you to try the following in any JavaScript console:
```
const testArray = [];
testArray.push(1)
```
And tell me if it breaks, or not, and why.